### PR TITLE
Show the curl request when possible.

### PR DESCRIPTION
**Pull request by pinarx**

### CODE REVIEW

File: tests/cases/push-generator.js [hunk 1]
```diff
@@ -674,15 +677,28 @@ PushGenerator.prototype.displayMessage = function() {
...
       headers.push(headerName + ': ' + request.headers[headerName]);
+      curlCommand += ' --header "' + headerName + ': ' + request.headers[headerName] + '"';
     });
 
     content.querySelector('#endpoint').textContent = request.url;
     content.querySelector('#headers').textContent = headers.join('\n');
     content.querySelector('#body').textContent = request.body;
 
+    if (request.body instanceof ArrayBuffer) {
+      curlError = 'Since web push protocol requires a stream as ' +
```
Comment (beverloo) [inline tests/cases/push-generator.js]: nit: `var`

File: tests/cases/push-generator.js [hunk 2]
```diff
@@ -674,15 +677,28 @@ PushGenerator.prototype.displayMessage = function() {
...
+          'the body of the request, there is no CURL command that will ' +
+          'stream an encrypted payload.';
+      content.querySelector('#error').textContent = curlError;
+    } else {
+      // content-length is needed for the curl request.
+      curlCommand += ' --header "Content-Length: 0"';
+      content.querySelector('#request').textContent = curlCommand;
```
Comment (beverloo) [inline tests/cases/push-generator.js]: I don't think there's an element having ID `request`. Is there a HTML part to this change?

File: tests/cases/push-generator.js (hunk 1 above)
Comment (pinarx) [inline tests/cases/push-generator.js]: done

File: tests/cases/push-generator.js (hunk 2 above)
Comment (pinarx) [inline tests/cases/push-generator.js]: ups, it somehow didn't go through - should be there now.

File: tests/cases/push-generator.js [hunk 3]
```diff
@@ -674,15 +677,28 @@ PushGenerator.prototype.displayMessage = function() {
...
 
     content.querySelector('#endpoint').textContent = request.url;
     content.querySelector('#headers').textContent = headers.join('\n');
     content.querySelector('#body').textContent = request.body;
 
+    if (request.body instanceof ArrayBuffer) {
+      var curlError = 'Since web push protocol requires a stream as ' +
+          'the body of the request, there is no CURL command that will ' +
+          'stream an encrypted payload.';
+      content.querySelector('#error').textContent = curlError;
```
Comment (beverloo) [inline tests/cases/push-generator.js]: It seems to me that either #error or #request would now always be presented as "[undefined]", which is a bit unfortunate. How about removing #error and hiding #request by default (through CSS `display: none`), and making it visible when we can populate the CURL request in the case below?

(I'm happy to do this in a follow-up :))

Comment (pinarx) [inline tests/cases/push-generator.js]: done, take a look

**Comment by beverloo** [comment]
Thank you!

**Pull request merged**